File: src/226_InvertBinaryTree.java
class InvertBinaryTree {
  public TreeNode invertTree(TreeNode root) {
    dfs(root);
    return root;
  }

  private void dfs(TreeNode root) {
    if (root==null) return;
    TreeNode temp;
    temp = root.left;
    root.left = root.right;
    root.right = temp;
    dfs(root.left);
    dfs(root.right);
  }
}

File: src/226_InvertBinaryTree.cpp
#include <local_leetcode.hpp>

class Solution {
  public:
    TreeNode* invertTree(TreeNode* root) {
      if (root) {
        TreeNode* tmp = root->left;
        root->left = invertTree(root->right);
        root->right = invertTree(tmp);
      }
      return root;
    }
};

int main() {
  EXECS(Solution::invertTree);
  return 0;
}
